### DISCUSSION TITLE
termux armv7  uname -a show Android

### DISCUSSION THREAD

**Discussion by szllzs** [termux-pacman/glibc-packages#35] [Q&A] [answered]
grun --shell
uname -a
the architecture displayed is incorrect,it shows android as not being gnu/linux

**Comment by Maxython** [comment]
The `uname` command is provided in the `coreutils` package from [termux/termux-packages](https://github.com/termux/termux-packages), it is based on the `bionic` library and does not depend on `glibc` in any way. The fact that it shows that this is `Android` and not `gnu-linux` is fine, but that it shows that you have an `armv7` architecture and not `aarch64` is already strange.

Try updating all packages or reinstalling `coreutils`. If this does not solve the problem, then create an **issue** in [termux/termux-packages](https://github.com/termux/termux-packages/issues).

**Comment by szllzs** [comment]
> The `uname` command is provided in the `coreutils` package from [termux/termux-packages](https://github.com/termux/termux-packages), it is based on the `bionic` library and does not depend on `glibc` in any way. The fact that it shows that this is `Android` and not `gnu-linux` is fine, but that it shows that you have an `armv7` architecture and not `aarch64` is already strange.
> 
> Try updating all packages or reinstalling `coreutils`. If this does not solve the problem, then create an **issue** in [termux/termux-packages](https://github.com/termux/termux-packages/issues).

![Screenshot_2023-06-15-21-55-36-54_84d3000e3f4017145260f7618db1d683](https://github.com/termux-pacman/glibc-packages/assets/13085785/6d8aa290-073b-49eb-8c84-746727462068)

**Reply by Maxython** [reply, marked as answer]
Now everything is clear, you have `aarch64` architecture but you have installed coreutils for `arm` (i.e. `armv7`). It has determined that your architecture is `arm64` (this is what the `armv8` part says) and that it supports running 32-bit programs (this is what the `l` part says).

[More](https://stackoverflow.com/questions/73688696/what-is-my-architecture-and-what-does-armv8l-exactly-means#:~:text=Generally%20arm%20or%20aarch32%20refers,mode%20would%20say%20arm64%20here.)

**Reply by szllzs** [reply]
If the device type is detected as Android in the shell of grun, it can lead to compilation errors when building some GNU software due to the definition of the ANDROID macro. In theory, after entering the shell of grun, the system type obtained should be GNU/Linux, and this works fine in the Arch package. However, the system type obtained is incorrect for the armv7 package.

**Reply by szllzs** [reply]
![无标题](https://github.com/termux-pacman/glibc-packages/assets/13085785/39e78a32-38c6-422f-b42d-2de839b2b83f)

**Reply by Maxython** [reply]
> If the device type is detected as Android in the shell of grun, it can lead to compilation errors when building some GNU software due to the definition of the ANDROID macro. In theory, after entering the shell of grun, the system type obtained should be GNU/Linux, and this works fine in the Arch package. However, the system type obtained is incorrect for the armv7 package.

Again, I repeat, you have `coreutils` based on the `bionic` library and the fact that it determines that you have `Android` is correct. `grun` the shell configures access to glibc commands and configures the shell itself to make them work, that's its whole job.

The only way to have `GNU/Linux` in `uname` is to compile `coreutils` against glibc.

**Comment by szllzs** [comment]
> Now everything is clear, you have `aarch64` architecture but you have installed coreutils for `arm` (i.e. `armv7`). It has determined that your architecture is `arm64` (this is what the `armv8` part says) and that it supports running 32-bit programs (this is what the `l` part says).
> 
> [More](https://stackoverflow.com/questions/73688696/what-is-my-architecture-and-what-does-armv8l-exactly-means#:~:text=Generally%20arm%20or%20aarch32%20refers,mode%20would%20say%20arm64%20here.)



> Now everything is clear, you have `aarch64` architecture but you have installed coreutils for `arm` (i.e. `armv7`). It has determined that your architecture is `arm64` (this is what the `armv8` part says) and that it supports running 32-bit programs (this is what the `l` part says).
> 
> [More](https://stackoverflow.com/questions/73688696/what-is-my-architecture-and-what-does-armv8l-exactly-means#:~:text=Generally%20arm%20or%20aarch32%20refers,mode%20would%20say%20arm64%20here.)
